src/NovelRT/DotNet/RuntimeService.cpp

```diff
@@ -132,7 +132,7 @@ namespace NovelRT::DotNet {
       if (result != 0)
       {
         _logger.logError("Failed to locate the specified managed function: {}", result);
-        throw std::runtime_error("Failed to locate the specified managed function");
+        throw Exceptions::FunctionNotFoundException("Failed to locate the specified managed function", result);
```

Comment:
I feel like the first argument of this exception could just be the default error that it appends the result to internally.